### PR TITLE
Only read layouts from source_dir or theme_dir

### PR DESCRIPTION
**Pull request by ashmaroli**
Reference: #6585 

With this Jekyll will no longer try to read layouts from `Dir.pwd` unless that path is the site's `source`

### CODE REVIEW

**Comment by ashmaroli** [comment]
### How Jekyll handles includes:
https://github.com/jekyll/jekyll/blob/fc28a3d8d19080ca37249428eae335fc44a923cf/lib/jekyll/site.rb#L445-L449

### How Jekyll looks for plugins:
https://github.com/jekyll/jekyll/blob/fc28a3d8d19080ca37249428eae335fc44a923cf/lib/jekyll/plugin_manager.rb#L92-L101

**Comment by pathawks** [comment]
This _isn't_ an optimization (as far as performance) so much as a simplification of behavior, correct?

**Comment by ashmaroli** [comment]
@pathawks, I think @Dirty considered this to be optimization as a consequence of the simplification..

**Comment by DirtyF** [comment]
We still need a review from someone @jekyll/build on this. 👀

File: test/test_layout_reader.rb [hunk 1]
```diff
@@ -27,8 +27,9 @@ class TestLayoutReader < JekyllUnitTest
         allow(Dir).to receive(:pwd).and_return(source_dir("blah"))
       end
 
-      should "know to use the layout directory relative to CWD" do
-        assert_equal LayoutReader.new(@site).layout_directory, source_dir("blah/_layouts")
+      should "ignore the layout directory in CWD and use the directory relative to site source" do
```
[the remark this answers is not in the document]
Comment (ashmaroli) [reply test/test_layout_reader.rb]: Yes @Crunch09 `CWD` stands for `current working directory`..
I think I should make it clearer for everyone..

**Comment by oe** [comment]
I don't remember being in `jekyll/build`, but whatever.

@jekyllbot: merge +major

**Pull request merged**